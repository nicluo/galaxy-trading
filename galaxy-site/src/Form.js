import React from 'react';

export const Form = () => (
  <div className="container my-4">
    <div className="form-group">
      <input type="text" className="form-control form-control-lg" id="query" aria-describedby="query" placeholder="how much is pish pish Iron ?" autoFocus="" />
    </div>
  </div>
);

export default Form;
