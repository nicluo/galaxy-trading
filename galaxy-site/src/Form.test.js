import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import Form from './Form';

describe('Form', () => {
  it('renders <Form /> component', () => {
    const wrapper = shallow(<Form />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('updates input value based on keypresses', () => {
    const wrapper = shallow(<Form />);
    const input = wrapper.find('input');
    input.simulate('change', {target: {value: 'test input'}});
    expect(wrapper.instance().state.input).toEqual('test input');
    expect(wrapper.find('input').props().value).toEqual('test input');
  });

  it('calls onSubmit when form is submitted', () => {
    const spy = sinon.spy();
    const wrapper = shallow(<Form onSubmit={spy} />);
    wrapper.find('input').simulate('change', {target: {value: 'test input'}});
    wrapper.find('form').simulate('submit', {preventDefault: () => {}});
    expect(spy.getCall(0).args[0]).toEqual('test input');
  });
});