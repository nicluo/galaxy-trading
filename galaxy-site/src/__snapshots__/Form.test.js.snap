// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders <Form /> component 1`] = `
<div
  className="container my-4"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        aria-describedby="query"
        autoFocus=""
        className="form-control form-control-lg"
        id="query"
        onChange={[Function]}
        placeholder="how much is pish pish Iron ?"
        type="text"
        value=""
      />
    </div>
  </form>
</div>
`;
